js-envs/src/main/scala/org/scalajs/jsenv/rhino/RhinoJSEnv.scala

```diff
@@ -301,32 +303,33 @@ final class RhinoJSEnv private (
   /** Loads the classpath. Either through lazy loading or by simply inserting */
   private def loadClasspath(context: Context, scope: Scriptable,
       classpath: CompleteClasspath): Unit = classpath match {
-    case cp: IRClasspath =>
+    case cp: LinkingUnitClasspath =>
       // Setup lazy loading classpath and source mapper
-      val optLoader = if (cp.scalaJSIR.nonEmpty) {
-        val loader = new ScalaJSCoreLib(semantics, cp)
+      val loader = new ScalaJSCoreLib(semantics, cp.linkingUnit)
 
-        // Setup sourceMapper
-        if (sourceMap) {
-          val scalaJSenv = context.newObject(scope)
+      // Setup sourceMapper
+      if (sourceMap) {
+        val scalaJSenv = context.newObject(scope)
 
-          scalaJSenv.addFunction("sourceMapper", args => {
-            val trace = Context.toObject(args(0), scope)
-            loader.mapStackTrace(trace, context, scope)
-          })
+        scalaJSenv.addFunction("sourceMapper", args => {
+          val trace = Context.toObject(args(0), scope)
+          loader.mapStackTrace(trace, context, scope)
+        })
 
-          ScriptableObject.putProperty(scope, "__ScalaJSEnv", scalaJSenv)
-        }
-
-        Some(loader)
-      } else {
-        None
+        ScriptableObject.putProperty(scope, "__ScalaJSEnv", scalaJSenv)
       }
 
       // Load JS libraries
       cp.jsLibs.foreach(dep => context.evaluateFile(scope, dep.lib))
 
-      optLoader.foreach(_.insertInto(context, scope))
+      loader.insertInto(context, scope)
+
+    // Deprecated, auto-linking
+    case cp: IRClasspath if cp.scalaJSIR.nonEmpty =>
+      val linkingUnit = linkIRClasspath(cp, semantics)
+      val linkingUnitCP = new LinkingUnitClasspath(cp.jsLibs, linkingUnit,
+          cp.requiresDOM, None)
+      loadClasspath(context, scope, linkingUnitCP)
```

Comment:
Could you make an overload of `loadClasspath` that is typed to take a `LinkingUnitClasspath`. I'd like to not rely on dynamic typing for correctness...